x-pack/plugins/synthetics/server/synthetics_service/service_api_client.ts

```diff
@@ -55,22 +55,28 @@ export class ServiceAPIClient {
     this.server = server;
   }
 
-  getHttpsAgent(url: string) {
+  getHttpsAgent(targetUrl: string) {
+    const parsedTargetUrl = new URL(targetUrl);
+
+    const rejectUnauthorized = parsedTargetUrl.hostname !== 'localhost' || !this.server.isDev;
+    const baseHttpsAgent = new https.Agent({ rejectUnauthorized });
```

Comment:
i think this will prevent us from using basic auth in staging environment against qa or staging env. which is probably fine.

Reply:
Thanks for adding this. I think it's a good think to clarify:

-> We couldn't use basic auth on those environments anyway because the server itself (service) will refuse basic auth.

The only environment in which basic auth should be enabled is `dev`.

Therefore this doesn't change the behaviour of the client.